Content.Server/Animals/Components/EggLayerComponent.cs

```diff
@@ -9,44 +11,47 @@ namespace Content.Server.Animals.Components;
 ///     It also grants an action to players who are controlling these entities, allowing them to do it manually.
 /// </summary>
 
-[RegisterComponent]
+[RegisterComponent, Access(typeof(EggLayerSystem))]
 public sealed partial class EggLayerComponent : Component
 {
-    [DataField]
-    public EntProtoId EggLayAction = "ActionAnimalLayEgg";
+    /// <summary>
+    ///     The item that gets laid/spawned, retrieved from animal prototype.
+    /// </summary>
+    [DataField(required: true), ViewVariables(VVAccess.ReadWrite)]
+    public List<EntitySpawnEntry> EggSpawn = default!;
 
     /// <summary>
-    ///     The amount of nutrient consumed on update.
+    ///     Player action.
     /// </summary>
-    [DataField, ViewVariables(VVAccess.ReadWrite)]
-    public float HungerUsage = 60f;
+    [DataField]
+    public EntProtoId EggLayAction = "ActionAnimalLayEgg";
+
+    [DataField]
+    public SoundSpecifier EggLaySound = new SoundPathSpecifier("/Audio/Effects/pop.ogg");
 
     /// <summary>
     ///     Minimum cooldown used for the automatic egg laying.
     /// </summary>
     [DataField, ViewVariables(VVAccess.ReadWrite)]
-    public float EggLayCooldownMin = 60f;
+    public float EggLayCooldownMin = 45f;
 
     /// <summary>
     ///     Maximum cooldown used for the automatic egg laying.
     /// </summary>
     [DataField, ViewVariables(VVAccess.ReadWrite)]
-    public float EggLayCooldownMax = 120f;
+    public float EggLayCooldownMax = 90f;
 
     /// <summary>
-    ///     Set during component init.
+    ///     The amount of nutrient consumed on update.
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite)]
-    public float CurrentEggLayCooldown;
-
-    [DataField(required: true), ViewVariables(VVAccess.ReadWrite)]
-    public List<EntitySpawnEntry> EggSpawn = default!;
-
-    [DataField]
-    public SoundSpecifier EggLaySound = new SoundPathSpecifier("/Audio/Effects/pop.ogg");
-
-    [DataField]
-    public float AccumulatedFrametime;
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public float HungerUsage = 25f;
 
     [DataField] public EntityUid? Action;
+
+    /// <summary>
+    ///     When to next try to produce.
+    /// </summary>
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer)), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan NextGrowth = TimeSpan.FromSeconds(0);
```

Comment:
This should use AutoPausedField with AutoGenerateComponentPause on the component.

Reply:
only EggLayerComponent or the others as well?

Reply:
The others too, yeah.